agent/pkg/server/server.go

```diff
@@ -390,18 +390,24 @@ func (s *Server) Preview(in *pb.PreviewRequest, srv pb.AutomationService_Preview
 		}
 	}()
 
+	var stackLocked bool
 	res, err := stack.Preview(ctx, opts...)
 	if err != nil {
-		s.log.Errorw("preview completed with an error", zap.Error(err))
-		return err
+		// If the error occurs due to a locked stacked, we need to surface this to the client.
+		if !strings.Contains(err.Error(), "Conflict: Another update is currently in progress") {
+			s.log.Errorw("preview completed with an error", zap.Error(err))
+			return err
+		}
+		stackLocked = true
```

Comment:
The `res` variable is not really usable if the preview failed, so we should return here.  Can we produce an error struct with an error code like `Locked`, that the client can unmarshal?